darkmoon/api/backend/endpoint.py

```diff
@@ -1,68 +1,151 @@
+    )
+
+
+@router.post(
+    "/metadata",
+    responses={
+        400: {"Client Error Response": "Bad Request"},
+    },
+)
 async def get_metadata_endpoint(
-    file: PyPath = Query(..., description="Path to the file"),
-    source_iso: PyPath = Query(..., description="Path to the source ISO"),
+    file: UploadFile = File(...),
+    source_iso: UploadFile = File(...),
 ) -> dict[str, Any]:
     """Get metadata."""
-    return get_metadata(file, source_iso.name)
+    with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
+        tmpfile.write(await file.read())
+        tmp_path = PyPath(tmpfile.name)
 
+    with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
+        tmpfile.write(await source_iso.read())
+        iso_path = str(PyPath(tmpfile.name))
+
+    return utils.get_metadata(tmp_path, iso_path)
 
-@router.get("/get-file-type")
+
+@router.post(
+    "/get-file-type",
+    responses={
+        400: {"Client Error Response": "Bad Request"},
+    },
+)
 async def get_file_type_endpoint(
-    file: PyPath = Query(..., description="Path to the file"),
+    file: UploadFile = File(...),
 ) -> str:
     """Get file."""
-    return get_file_type(file)
+    with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
+        tmpfile.write(await file.read())
+        tmp_path = PyPath(tmpfile.name)
+    return utils.get_file_type(tmp_path)
 
 
-@router.get("/get-hash")
+@router.post(
+    "/get-hash",
+    responses={
+        400: {"Client Error Response": "Bad Request"},
+    },
+)
 async def get_hashes_endpoint(
-    file: PyPath = Query(..., description="Path to the file"),
+    file: UploadFile = File(...),
 ) -> dict[str, str]:
     """Get hashes of files."""
-    return get_hashes(file)
+    with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
+        tmpfile.write(await file.read())
+        tmp_path = PyPath(tmpfile.name)
+    return utils.get_hashes(tmp_path)
 
 
 @router.post("/get-all-exe-metadata", response_class=JSONResponse)
 async def get_all_exe_metadata_endpoint(
-    file: PyPath = Query(..., description="Path to the file"),
+    file: UploadFile = File(...),
 ) -> dict[str, Any]:
     """Post the exe metadata."""
-    return get_all_exe_metadata(file)
+    with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
+        tmpfile.write(await file.read())
+        tmp_path = PyPath(tmpfile.name)
+    return utils.get_all_exe_metadata(tmp_path)
 
 
-@router.post("/extract_file")
+@router.post(
+    "/extract-files",
+    responses={
+        400: {"Client Error Response": "Bad Request"},
+        422: {"Client Error Response": "Unprocessable Content"},
+        500: {"Server Error Response": "Internal Server Error"},
+        504: {"Server Error Response": "Gateway Timeout"},
+    },
+)
 async def extract_files_endpoint(
-    file: UploadFile = File(...),
+    file: UploadFile = File(...),  # only vmdks
     source_iso: PyPath = PyPath("..."),
```

Comment:
How come some parameters for source_iso take in a path while other take UploadFile?